teerex/src/lib.rs

```diff
@@ -717,11 +597,45 @@ impl<T: Config> Pallet<T> {
 			Err(<Error<T>>::RemoteAttestationTooOld.into())
 		}
 	}
-}
 
-impl<T: Config> OnTimestampSet<T::Moment> for Pallet<T> {
-	fn on_timestamp_set(moment: T::Moment) {
-		Self::unregister_silent_workers(moment)
+	pub fn poke_shard(
```

Comment:
From the outside it is a bit hard to guess, what is `poke`, I think the more commonly used term `touch` reflects better, what this function represents.

Reply:
Thanks for the intermediate review. I'll iterate and guess its better to wait with the next phase of the review